lib/formats/bs_format_ripejson.c

```diff
@@ -0,0 +1,679 @@
+/* =============================================================== */
+/* ==================== PUBLIC API BELOW HERE ==================== */
+/* =============================================================== */
+/* =============================================================== */
+
+int bs_format_ripejson_create(bgpstream_format_t *format,
+                         bgpstream_resource_t *res)
+{
+  BS_FORMAT_SET_METHODS(ripejson, format);
+
+  if ((format->state = malloc_zero(sizeof(state_t))) == NULL) {
+    return -1;
+  }
+
+  parsebgp_opts_init(&STATE->opts);
+  bgpstream_parsebgp_opts_init(&STATE->opts);
+  STATE->opts.bgp.marker_omitted = 1;
+  STATE->opts.bgp.asn_4_byte= 1;
+
+  return 0;
+}
+
+bgpstream_format_status_t
+bs_format_ripejson_populate_record(bgpstream_format_t *format,
+                              bgpstream_record_t *record)
+{
+  int newread, rc;
+
+  newread = bgpstream_transport_readline(format->transport, &STATE->json_string_buffer, BGPSTREAM_PARSEBGP_BUFLEN);
+
+  assert(newread < BGPSTREAM_PARSEBGP_BUFLEN);
+
+  if (newread <0 ) {
+    record -> status = BGPSTREAM_RECORD_STATUS_CORRUPTED_RECORD;
+    record -> collector_name [0] = '\0';
+    return BGPSTREAM_FORMAT_CORRUPTED_DUMP;
+  } else if ( newread == 0 ){
+    return BGPSTREAM_FORMAT_END_OF_DUMP;
+  }
+
+  if( ( rc = bs_format_process_json_fields(format, record) ) != 0){
+    return rc;
+  }
+
+  return BGPSTREAM_FORMAT_OK;
+}
+
+int bs_format_ripejson_get_next_elem(bgpstream_format_t *format,
+                                bgpstream_record_t *record,
+                                bgpstream_elem_t **elem)
+{
+  int rc = 0;
+
+  if (RDATA == NULL || RDATA->end_of_elems != 0) {
+    // end-of-elems
+    return 0;
+  }
+
+  switch(RDATA->msg_type){
+  case RIPE_JSON_MSG_TYPE_ANNOUNCE:
+  case RIPE_JSON_MSG_TYPE_WITHDRAW:
+    rc = bgpstream_parsebgp_process_update(&RDATA->upd_state, RDATA->elem,
+                                               RDATA->msg->types.bgp);
+    if (rc <= 0) {
+      return rc;
+    }
+    break;
+  case RIPE_JSON_MSG_TYPE_STATUS:
+    RDATA->elem->type = BGPSTREAM_ELEM_TYPE_PEERSTATE;
+    if(RDATA->status_msg_state == 0){
+      // "down" state
+      RDATA->elem->old_state = BGPSTREAM_ELEM_PEERSTATE_UNKNOWN;
+      RDATA->elem->new_state = BGPSTREAM_ELEM_PEERSTATE_IDLE;
+    } else if (RDATA->status_msg_state == 1  ){
+      // "connected" state
+      RDATA->elem->old_state = BGPSTREAM_ELEM_PEERSTATE_UNKNOWN;
+      RDATA->elem->new_state = BGPSTREAM_ELEM_PEERSTATE_ESTABLISHED;
+    } else {
+      RDATA->elem->old_state = BGPSTREAM_ELEM_PEERSTATE_UNKNOWN;
+      RDATA->elem->new_state = BGPSTREAM_ELEM_PEERSTATE_UNKNOWN;
+      fprintf(stderr, "WARN: unsupported status type, %d\n", RDATA->status_msg_state);
+    }
+    RDATA->end_of_elems = 1;
+    rc = 1;
+    break;
+  case RIPE_JSON_MSG_TYPE_OPEN:
+    RDATA->elem->type = BGPSTREAM_ELEM_TYPE_PEERSTATE;
+    if(RDATA->open_msg_direction == 0){
```

Comment:
and here

Reply:
The changes for the two comments above, the arrow space change, and removing of a comment section:
https://github.com/CAIDA/libbgpstream/commit/e9e84cafac15aa187fabce4ca2057ae8c11e64ca